app/services/make_an_offer.rb

```diff
@@ -0,0 +1,15 @@
+class MakeAnOffer
+  Response = Struct.new(:successful?, :application_choice)
+
+  def initialize(application_choice:, offer_conditions:)
+    @application_choice = application_choice
+    @offer_conditions = offer_conditions
+  end
+
+  def call
+    @application_choice.status = @offer_conditions.present? ? :conditional_offer : :unconditional_offer
+    @application_choice.offer = @offer_conditions.present? ? @offer_conditions : { 'conditions' => [] }
+    @application_choice.save
```

Comment:
This means that the `NotImplementedError` in the controller will never be reached. Perhaps check i `@application_choice.save` succeeds, and return `Response.new(false @application_choice)`. As it stands now it'll always return a success.

Alternatively, make this a `save!` and don't check the result in the controller.

Reply:
As there is no logic here anyway this code will need to be revisited when we do https://trello.com/c/RzDNYCQw/83-implement-business-rules-around-status-changes. I think it's better to remove the `NotImplementedError` and let the `true` stand, so we don't even gesture towards the unhappy path